extras/indexing/src/main/java/mvm/rya/indexing/FilterFunctionOptimizer.java

```diff
@@ -101,23 +102,23 @@ public void setConf(final Configuration conf) {
         init = false;
         try {
             init();
-        } catch (final NumberFormatException | UnknownHostException e) {
+        } catch (final NumberFormatException | IOException e) {
             LOG.error("Unable to update to use new config, falling back to the old config.", e);
             init = true;
         }
     }
 
-    private synchronized void init() throws NumberFormatException, UnknownHostException {
+    private synchronized void init() throws NumberFormatException, IOException {
         if (!init) {
             if (ConfigUtils.getUseMongo(conf)) {
                 try {
                     final MongoClient mongoClient = MongoConnectorFactory.getMongoClient(conf);
-                    geoIndexer = new MongoGeoIndexer(mongoClient);
-                    geoIndexer.setConf(conf);
-                    freeTextIndexer = new MongoFreeTextIndexer(mongoClient);
-                    freeTextIndexer.setConf(conf);
-                    temporalIndexer = new MongoTemporalIndexer(mongoClient);
-                    temporalIndexer.setConf(conf);
+                    geoIndexer = new MongoGeoIndexer();
+                    ((AbstractMongoIndexer) geoIndexer).initIndexer(conf, mongoClient);
```

Comment:
GeoIndexer has null fields: storageStrategy , predicates.  The implementation init is not called?  In the code I see from develop, init does this 
<code>predicates = ConfigUtils.getGeoPredicates(conf);  
storageStrategy = ...<code>